File: app/src/main/java/com/nuhkoca/offlineapp/binding/ListBindingAdapter.java
package com.nuhkoca.offlineapp.binding;

import com.nuhkoca.offlineapp.base.BaseAdapter;
import com.nuhkoca.offlineapp.repository.Resource;

import java.util.List;

import androidx.databinding.BindingAdapter;
import androidx.recyclerview.widget.RecyclerView;

public class ListBindingAdapter {

    public ListBindingAdapter() {
    }

    @SuppressWarnings("unchecked")
    @BindingAdapter(value = "android:resource")
    public void setResource(RecyclerView recyclerView, Resource resource) {
        RecyclerView.Adapter adapter = recyclerView.getAdapter();
        if (adapter == null)
            return;

        if (resource == null || resource.data == null)
            return;

        if (adapter instanceof BaseAdapter) {
            ((BaseAdapter) adapter).setData((List) resource.data);
        }
    }
}

File: app/src/main/java/com/nuhkoca/offlineapp/binding/ListBindingAdapter.kt
package com.nuhkoca.offlineapp.binding

import androidx.databinding.BindingAdapter
import androidx.recyclerview.widget.RecyclerView
import com.nuhkoca.offlineapp.base.BaseAdapter
import com.nuhkoca.offlineapp.repository.Resource

class ListBindingAdapter {

    @Suppress("UNCHECKED_CAST")
    @BindingAdapter(value = ["android:resource"])
    fun setResource(recyclerView: RecyclerView, resource: Resource<*>) {
        val adapter = recyclerView.adapter ?: return

        resource.data?.let {
            if (adapter is BaseAdapter<*, *>) {
                adapter.setData(it as List<Any>)
            }
        }
    }
}
